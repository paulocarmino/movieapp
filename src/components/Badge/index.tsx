export type BadgeProps = {
  children: JSX.Element | string
}

const Badge = ({ children }: BadgeProps) => (
  <span className="inline-flex items-center py-0.5 px-2.5 mx-1 text-xs font-medium text-gray-800 bg-gray-100 hover:bg-gray-300 rounded-full">
    {children}
  </span>
)

export default Badge
