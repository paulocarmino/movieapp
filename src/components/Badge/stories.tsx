import { Story, Meta } from '@storybook/react'
import Badge from '.'

export default {
  title: 'Badge',
  component: Badge,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  },
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => <Badge {...args}>Example</Badge>

Default.args = {
  children: 'Example'
}
