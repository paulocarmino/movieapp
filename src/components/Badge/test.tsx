import { render, screen } from '@testing-library/react'

import Badge from '.'

describe('<Badge />', () => {
  it('should render the component', () => {
    const { container } = render(<Badge>Example</Badge>)

    expect(screen.getByText('Example')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
