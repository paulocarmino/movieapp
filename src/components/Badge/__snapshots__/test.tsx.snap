// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render the component 1`] = `
<span
  class="inline-flex items-center py-0.5 px-2.5 mx-1 text-xs font-medium text-gray-800 bg-gray-100 hover:bg-gray-300 rounded-full"
>
  Example
</span>
`;
